main.py

```diff
@@ -126,36 +126,24 @@ def setup_target_task_training(args, target_tasks, model, strict):
         log.info("Loading existing model from %s...", best_path)
         load_model_state(model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict)
     else:
-        # Look for target train checkpoints (available only if we're restoring from a run that
-        # already finished), then look for training checkpoints.
-
-        best_path = get_best_checkpoint_path(args.run_dir, "target_train")
-        if best_path:
-            # Continue target task training.
-            load_model_state(
-                model, best_path, args.cuda, task_names_to_avoid_loading, strict=strict
-            )
-        else:
-            # Load a pretrained model and start target task training.
+        if args.do_pretrain == 1:
             best_pretrain = get_best_checkpoint_path(args.run_dir, "pretrain")
             if best_pretrain:
                 load_model_state(
                     model, best_pretrain, args.cuda, task_names_to_avoid_loading, strict=strict
                 )
-            else:
-                assert_for_log(
-                    args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
-                )
-
+        else:
             if args.transfer_paradigm == "finetune":
                 # We want to do target training without pretraining, thus
                 # we need to first create a checkpoint to come back to for each of
                 # the target tasks to finetune.
+                assert_for_log(
+                    args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
+                )
                 model_state = model.state_dict()
                 model_path = os.path.join(args.run_dir, "model_state_untrained_pre_target_train.th")
                 torch.save(model_state, model_path)
-
-            log.warning("Evaluating untrained encoder parameters!")
+            log.warning("Using untrained encoder parameters!")
```

Comment:
We can have some kind of warning here, but this is misleading if we're doing fine tuning.